lib/active_merchant/billing/gateways/realex.rb

```diff
@@ -147,14 +147,15 @@ def build_purchase_or_authorization_request(action, money, credit_card, options)
         xml.target!
       end
 
-      def build_capture_request(authorization, options)
+      def build_capture_request(money, authorization, options)
         timestamp = new_timestamp
         xml = Builder::XmlMarkup.new :indent => 2
         xml.tag! 'request', 'timestamp' => timestamp, 'type' => 'settle' do
           add_merchant_details(xml, options)
+          add_amount(xml, money, options)
           add_transaction_identifiers(xml, authorization, options)
           add_comments(xml, options)
-          add_signed_digest(xml, timestamp, @options[:login], sanitize_order_id(options[:order_id]), nil, nil, nil)
+          add_signed_digest(xml, timestamp, @options[:login], sanitize_order_id(options[:order_id]), amount(money), (options[:currency] || currency(money)), nil)
```

Comment:
`options[:currency] || currency(money)` is a pattern that shows up several times here (e.g. in `build_refund_request` and `build_purchase_or_authorization_request`). Do you think it might be worth extracting? (And that's a genuine question, because three appearances is enough to make me wonder, but not enough to be an obvious win.)

Reply:
Honestly I think the `currency` method should accept `options` and build in that conditional. It doesn't otherwise seem to be extracted for any other gateways, so I'm somewhat inclined to leave it as is.